internal/observability/metrics.go

```diff
@@ -11,59 +11,48 @@ import (
 	"github.com/sirupsen/logrus"
 	"github.com/supabase/auth/internal/conf"
 
-	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.opentelemetry.io/otel"
 	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
 	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
 	"go.opentelemetry.io/otel/exporters/prometheus"
 	"go.opentelemetry.io/otel/metric"
-	metricglobal "go.opentelemetry.io/otel/metric/global"
-	metricinstrument "go.opentelemetry.io/otel/metric/instrument"
-	basicmetriccontroller "go.opentelemetry.io/otel/sdk/metric/controller/basic"
-	exportmetricaggregation "go.opentelemetry.io/otel/sdk/metric/export/aggregation"
-	basicmetricprocessor "go.opentelemetry.io/otel/sdk/metric/processor/basic"
-	simplemetricselector "go.opentelemetry.io/otel/sdk/metric/selector/simple"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
 
 	otelruntimemetrics "go.opentelemetry.io/contrib/instrumentation/runtime"
 )
 
 func Meter(instrumentationName string, opts ...metric.MeterOption) metric.Meter {
-	return metricglobal.Meter(instrumentationName, opts...)
+	return otel.Meter(instrumentationName, opts...)
 }
 
-func ObtainMetricCounter(name, desc string) metricCounter {
```

Comment:
`Int64Counter` is now synchronous and `Int64ObservableCounter`'s [are async](https://pkg.go.dev/go.opentelemetry.io/otel/metric#Int64ObservableCounter)